runtime-modules/forum/src/lib.rs

```diff
@@ -1536,6 +1566,24 @@ decl_module! {
 }
 
 impl<T: Trait> Module<T> {
+    /// Hash poll description and poll alternatives descriptions, coverting `PollInput` into `Poll`
+    fn from_poll_input(poll_input: PollInput<T::Moment>) -> Poll<T::Moment, T::Hash> {
```

Comment:
Could we implement  `From<PollInput>` trait instead?

Reply:
We can, i even thought about this approach, but i don`t like that it would require to pass a hash function as additional generic parameter in that case.

Reply:
We use an additional generic parameter anyway (in the result type) and the usage of the `From<>` implementation is as easy as `let poll = poll_input.clone().map(|pi| pi.into());` . However, you use `T::calculate_hash()` inside of the conversion function and I don't see an easy way to use it inside of the From-trait implementation. Let's keep your initial approach.